### PR TITLE
Fix clienttime, error throwing and linter happy, also revert some changes after types

### PR DESCRIPTION
**Pull request by martenrichter**

### CODE REVIEW

**Comment by martenrichter** [comment]
Nice there is a seg fault in CI/CD but not in my local machine (Release vs. Debug)

**Comment by martenrichter** [comment]
Segfault still there, but only in release builds....

**Comment by martenrichter** [comment]
TODO split delegate from client

File: lib/client.js [hunk 1]
```diff
@@ -94,7 +93,7 @@ export class Http3Client extends Http3WebTransport {
   /**
    * @param {ClientConnectedEvent} args
    */
-  onClientConnected (args) {
```
Comment (achingbrain) [inline lib/client.js:97]: Is it worth adding a linter for this?  [standard](https://www.npmjs.com/package/standard) or something?

File: lib/session.js [hunk 2]
```diff
@@ -395,6 +399,7 @@ export class Http3WTSession {
           visitor.onClose(args)
           break
         case 'DatagramReceived':
+          // eslint-disable-next-line no-prototype-builtins
           if (visitor && args.hasOwnProperty('datagram'))
```
Comment (achingbrain) [inline lib/session.js:403]: This should pacify the linter:

```suggestion
          if (visitor && Object.prototype.hasOwnProperty.call(args, 'datagram'))
```

File: lib/session.js [hunk 3]
```diff
@@ -404,7 +409,9 @@ export class Http3WTSession {
         case 'Http3WTStreamVisitor':
           if (
             visitor &&
+            // eslint-disable-next-line no-prototype-builtins
             args.hasOwnProperty('bidirectional') &&
```
Comment (achingbrain) [inline lib/session.js:413]:
```suggestion
            Object.prototype.hasOwnProperty.call(args, 'bidirectional') &&
```

File: lib/session.js [hunk 4]
```diff
@@ -404,7 +409,9 @@ export class Http3WTSession {
...
+            // eslint-disable-next-line no-prototype-builtins
             args.hasOwnProperty('incoming')
```
Comment (achingbrain) [inline lib/session.js:415]:
```suggestion
            Object.prototype.hasOwnProperty.call(args, 'incoming')
```

File: lib/session.js (hunk 2 above)
Comment (achingbrain) [inline lib/session.js:403]: This should pacify the linter:

```suggestion
          if (visitor && Object.prototype.hasOwnProperty.call(args, 'datagram'))
```

File: lib/stream.js [hunk 5]
```diff
@@ -295,6 +299,7 @@ export class Http3WTStream {
           visitor.onStreamRecvSignal(args)
           break
         case 'StreamRead':
+          // eslint-disable-next-line no-prototype-builtins
           if (visitor && args.hasOwnProperty('data')) {
```
Comment (achingbrain) [inline lib/stream.js:303]:
```suggestion
          // eslint-disable-next-line no-prototype-builtins
          if (visitor && Object.prototype.hasOwnProperty.call(args, 'data')) {
```

File: lib/client.js (hunk 1 above)
Comment (martenrichter) [reply lib/client.js:97]: There should be already some linter config files in the project. So far I do not protect the branch with the linter, but I always check against it manually before committing.

File: lib/session.js (hunk 3 above)
Comment (martenrichter) [reply lib/session.js:413]: Sure this will help

File: lib/stream.js (hunk 5 above)
Comment (martenrichter) [reply lib/stream.js:303]: Prefect

**Comment by martenrichter** [comment]
Should work now, after this the test PR has to be rebased.

File: lib/client.js (hunk 1 above)
Comment (achingbrain) [reply lib/client.js:97]: Might be worth adding an npm script for it and running it in CI.  Will submit a PR.

Comment (martenrichter) [reply lib/client.js:97]: Ok, good idea, had just not a high priority, while I was alone.

Comment (achingbrain) [reply lib/client.js:97]: Completely fair!

**Pull request merged**